cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -538,6 +542,720 @@ void accumulate_edge_results(
+      rmm::device_uvector<std::byte> d_tmp_storage(0, handle.get_stream());
+
+      // Process each chunk
+      for (size_t chunk_i = 0; chunk_i < num_chunks; ++chunk_i) {
+        size_t chunk_vertex_start    = h_vertex_chunk_offsets[chunk_i];
+        size_t chunk_vertex_end      = h_vertex_chunk_offsets[chunk_i + 1];
+        size_t chunk_distance_start  = h_distance_level_chunk_offsets[chunk_i];
+        size_t chunk_distance_end    = h_distance_level_chunk_offsets[chunk_i + 1];
+        size_t chunk_size            = chunk_vertex_end - chunk_vertex_start;
+        size_t num_segments_in_chunk = chunk_distance_end - chunk_distance_start;
+
+        printf("DEBUG: Processing chunk %zu: %zu vertices, %zu distance levels\n",
+               chunk_i,
+               chunk_size,
+               num_segments_in_chunk);
+
+        // Gather data for this chunk from original bucket arrays
+        size_t write_offset = 0;
+        std::vector<size_t> chunk_segment_offsets;
+        chunk_segment_offsets.push_back(0);
+
+        // Map distance level indices to actual distance values
+        std::vector<vertex_t> distance_levels_in_chunk;
+        for (vertex_t d = 1; d <= global_max_distance; ++d) {
+          if (d < distance_buckets_vertices.size() && distance_buckets_vertices[d].size() > 0) {
+            distance_levels_in_chunk.push_back(d);
+          }
+        }
+
+        // Copy data for distance levels in this chunk
+        for (size_t level_idx = chunk_distance_start; level_idx < chunk_distance_end; ++level_idx) {
+          vertex_t d        = distance_levels_in_chunk[level_idx];
+          size_t level_size = distance_buckets_vertices[d].size();
+
+          // Copy vertices with int32_t conversion
+          thrust::transform(handle.get_thrust_policy(),
+                            distance_buckets_vertices[d].begin(),
+                            distance_buckets_vertices[d].end(),
+                            chunk_vertices_int32.begin() + write_offset,
+                            [] __device__(vertex_t v) { return static_cast<int32_t>(v); });
+
+          // Copy sources
+          thrust::copy(handle.get_thrust_policy(),
+                       distance_buckets_sources[d].begin(),
+                       distance_buckets_sources[d].end(),
+                       chunk_sources.begin() + write_offset);
+
+          write_offset += level_size;
+          chunk_segment_offsets.push_back(write_offset);
+        }
+
+        if (num_segments_in_chunk > 0) {
+          // Copy segment offsets to device
+          rmm::device_uvector<size_t> d_chunk_segment_offsets(chunk_segment_offsets.size(),
+                                                              handle.get_stream());
+          raft::update_device(d_chunk_segment_offsets.data(),
+                              chunk_segment_offsets.data(),
+                              chunk_segment_offsets.size(),
+                              handle.get_stream());
```

Comment:
This too.

See this.
https://github.com/rapidsai/cugraph/blob/branch-25.10/cpp/src/structure/detail/structure_utils.cuh#L508

You can just use distance_level_offsets (just need to shift the offset values properly).